packages/babel-plugin-transform-typescript/src/index.js

```diff
@@ -190,10 +144,63 @@ export default declare((api, { jsxPragma = "React" }) => {
         // We do this here instead of in a `ClassProperty` visitor because the class transform
         // would transform the class before we reached the class property.
         path.get("body.body").forEach(child => {
-          if (child.isClassProperty()) {
-            child.node.typeAnnotation = null;
+          const childNode = child.node;
+
+          if (t.isClassMethod(childNode) && childNode.kind === "constructor") {
+            // Collect parameter properties
+            const parameterProperties = [];
+            for (const param of childNode.params) {
+              if (
+                param.type === "TSParameterProperty" &&
+                !PARSED_PARAMS.has(param.parameter)
+              ) {
+                PARSED_PARAMS.add(param.parameter);
```

Comment:
Mostly out of curiosity - it looks like this is the only real functional change added to this block, to prevent duplicate parameters from being added to the `parameterProperties` list?  What sort of case is this preventing?  How would two different entries of `childNode.params` have the same `parameter` object?

Reply:
It requires a bit of understanding of the class transformation:
https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-classes/src/index.js#L41

Ultimately, it prevents params from getting parsed twice.